--- socket.c	2019-07-02 12:27:14.000000000 +0000
+++ socket.c	2019-09-03 15:31:24.000000000 +0000
@@ -366,7 +366,7 @@
 	struct socket *new4 = NULL, *new6 = NULL;
 	struct udp_port_cfg port4 = {
 		.family = AF_INET,
-		.local_ip.s_addr = htonl(INADDR_ANY),
+//		.local_ip.s_addr = htonl(INADDR_ANY),
 		.local_udp_port = htons(port),
 		.use_udp_checksums = true
 	};
@@ -374,7 +374,7 @@
 	int retries = 0;
 	struct udp_port_cfg port6 = {
 		.family = AF_INET6,
-		.local_ip6 = IN6ADDR_ANY_INIT,
+//		.local_ip6 = IN6ADDR_ANY_INIT,
 		.use_udp6_tx_checksums = true,
 		.use_udp6_rx_checksums = true,
 		.ipv6_v6only = true
